lib/cdo/web_purify.rb

```diff
@@ -1,46 +1,106 @@
+require 'net/http'
 require 'open-uri'
 require 'json'
 require 'dynamic_config/gatekeeper'
 require_relative '../../pegasus/src/env'
 
 module WebPurify
+  # WebPurify limits us to 30,000 characters per request and 4 simultaneous requests per API key
+  API_ENDPOINT = URI('http://api1.webpurify.com/services/rest').freeze
+  CHARACTER_LIMIT = 30_000
+  REQUEST_LIMIT = 4
+  CONNECTION_OPTIONS = {
+    read_timeout: DCDO.get('webpurify_http_read_timeout', 10),
+    open_timeout: DCDO.get('webpurify_tcp_connect_timeout', 5)
+  }
   ISO_639_1_TO_WEBPURIFY = {
     'es' => 'sp',
     'ko' => 'kr',
     'ja' => 'jp'
   }.freeze
 
+  # Note: If text has a string of text without whitespace longer than CHARACTER_LIMIT,
+  # the entire substring will count as one long chunk and be given its own request to WebPurify.
+  def self.split_text(text, max_chunk_length = CHARACTER_LIMIT)
+    words = text.split(/\s/)
+    chunks = []
+    current_chunk = ""
+
+    words.each do |word|
+      if current_chunk.length + word.length + 1 <= max_chunk_length
+        current_chunk += " " unless current_chunk.empty?
+        current_chunk += word
+      else
+        chunks << current_chunk unless current_chunk.empty?
+        current_chunk = word
+      end
+    end
+
+    chunks << current_chunk unless current_chunk.empty?
+    chunks
+  end
+
   # Returns the all profanities in text (if any) or nil (if none).
   # @param [String] text The text to search for profanity within.
   # @param [Array[String]] language_codes The set of languages to search for profanity in.
   # @return [Array<String>, nil] The profanities (if any) or nil (if none).
   def self.find_potential_profanities(text, language_codes = ['en'])
     return nil unless CDO.webpurify_key && Gatekeeper.allows('webpurify', default: true)
     return nil if text.nil?
+
+    # This is an artificial limit to prevent us from profanity-checking a file up to 5MB (the project size limit)
```

Comment:
nit: should this comment read "prevent us from profanity-checking a file over 5MB"?

Reply:
Yeah I think I was confused by the "up to 5MB" line, does this limit have anything to do with 5 mb?

Reply:
No, it doesn't, except that that's the de factor limit without the URI limit. I could remove the reference to 5 MB and state that it's an arbitrary size limit. Do you think that would be clearer?

Reply:
ah ok I see. Yeah I think removing the 5 mb reference is clearer